src/probability_theory/stopping.lean

```diff
@@ -0,0 +1,271 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.function.conditional_expectation
+import probability_theory.notation
+
+/-!
+# Filtration and stopping time
+
+This file defines some standard definition from the theory of stochastic processes including
+filtrations and stopping times. These definitions are used to model the amount of information
+at a specific time and is the first step in formalizing stochastic processes.
+
+## Main definitions
+
+* `measure_theory.filtration`: a filtration on a measurable space
+* `measure_theory.adapted`: a sequence of functions `u` is said to be adapted to a
+  filtration `f` if at each point in time `i`, `u i` is `f i`-measurable
+* `measure_theory.filtration.natural`: the natural filtration with respect to a sequence of
+  measurable functions is the smallest filtration to which it is adapted to
+* `measure_theory.stopping_time`: a stopping time with respect to some filtration `f` is a
+  function `τ` such that for all `i`, the preimage of `{j | j ≤ i}` along `τ` is
+  `f i`-measurable
+* `measure_theory.stopping_time.measurable_space`: the σ-algebra associated with a stopping time
+
+## Tags
+
+filtration, stopping time, stochastic process
+
+-/
+
+noncomputable theory
+open_locale classical measure_theory nnreal ennreal topological_space probability_theory
+
+namespace measure_theory
+
+/-- A `filtration` on measurable space `α` with σ-algebra `m` is a monotone
+sequence of of sub-σ-algebras of `m`. -/
+structure filtration (α ι : Type*) [preorder ι] (m : measurable_space α) :=
+(seq : ι → measurable_space α)
+(mono : monotone seq) (le : ∀ i : ι, seq i ≤ m)
+
+instance {α ι : Type*} [preorder ι] {m : measurable_space α} :
+  has_coe_to_fun (filtration α ι m) (λ _, ι → measurable_space α) :=
+⟨λ f, f.seq⟩
+
+instance {α ι : Type*} [preorder ι] {m : measurable_space α} : inhabited (filtration α ι m) :=
+⟨⟨λ i, m, monotone_const, λ i, le_rfl⟩⟩
+
+open topological_space
+
+variables {α β ι : Type*} {m : measurable_space α} [measurable_space β]
```

Comment:
you could put those lines above the two instances and remove some of their arguments.